### DISCUSSION TITLE
Dependency Injection with Flagd

### DISCUSSION THREAD

**Discussion by tomflenner** [open-feature/dotnet-sdk#265] [Q&A] [answered]
Hi !

I can see there is a IFeatureClient interface on the .NET SDK.

Do the SDK provide a way to register this interface and use dependency injection instead of :
`Api.Instance.GetClient();`

I would like to write a custom attribute to protect api endpoint using open feature and flagd but I struggle a little bit with the SDK.

Thanks in advance ! 🙏🏻

**Comment by beeme1mr** [comment]
Hey @tomflenner, we're working on a [v2](https://github.com/open-feature/dotnet-sdk/milestone/2) of the .NET SDK that will make DI easier. FYI @askpt @toddbaert.

**Reply by tomflenner** [reply]
Hi @beeme1mr 👋 !

Thanks for the answer. Can't wait to see this v2 🔥.

**Comment by askpt** [comment, marked as answer]
Hey @tomflenner, the experimental version for the dependency injection was released as part of https://github.com/open-feature/dotnet-sdk/compare/v2.0.0...v2.1.0.
